diff --git a/bundle/ngx_lua-0.10.29R2/config b/bundle/ngx_lua-0.10.29R2/config
--- a/bundle/ngx_lua-0.10.29R2/config
+++ b/bundle/ngx_lua-0.10.29R2/config
@@ -26 +26 @@ if [ -n "$LUAJIT_INC" -o -n "$LUAJIT_LIB" ]; then
-        ngx_feature_libs="$ngx_lua_opt_L -llua51"
+        ngx_feature_libs="$ngx_lua_opt_L -lluajit-5.1"
diff --git a/bundle/ngx_stream_lua-0.0.17R4/config b/bundle/ngx_stream_lua-0.0.17R4/config
--- a/bundle/ngx_stream_lua-0.0.17R4/config
+++ b/bundle/ngx_stream_lua-0.0.17R4/config
@@ -26 +26 @@ if [ -n "$LUAJIT_INC" -o -n "$LUAJIT_LIB" ]; then
-        ngx_feature_libs="$ngx_lua_opt_L -llua51"
+        ngx_feature_libs="$ngx_lua_opt_L -lluajit-5.1"
diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -949 +949 @@ if ($opts->{stream}) {
-                    " CJSON_LDFLAGS=\"-shared -L$luajit_root -llua51\"";
+                    " CJSON_LDFLAGS=\"-shared -L$luajit_root -lluajit-5.1\"";
@@ -1010 +1010 @@ if ($opts->{stream}) {
-                $extra_opts .= " LDFLAGS=\"-shared -L$luajit_root -llua51\"";
+                $extra_opts .= " LDFLAGS=\"-shared -L$luajit_root -lluajit-5.1\"";
@@ -1046 +1046 @@ if ($opts->{stream}) {
-                $extra_opts .= " LDFLAGS=\"-shared -L$luajit_root -llua51\"";
+                $extra_opts .= " LDFLAGS=\"-shared -L$luajit_root -lluajit-5.1\"";
@@ -1097 +1097 @@ if ($opts->{stream}) {
-                $extra_opts .= " LDFLAGS=\"-shared -L$luajit_root -llua51\"";
+                $extra_opts .= " LDFLAGS=\"-shared -L$luajit_root -lluajit-5.1\"";
