diff --git a/src/ngx_http_vhost_traffic_status_dump.c b/src/ngx_http_vhost_traffic_status_dump.c
--- a/src/ngx_http_vhost_traffic_status_dump.c
+++ b/src/ngx_http_vhost_traffic_status_dump.c
@@ -6,6 +6,13 @@
 #include "ngx_http_vhost_traffic_status_module.h"
 #include "ngx_http_vhost_traffic_status_dump.h"
 
+#if (NGX_WIN32)
+#define ngx_lock_fd(fd) 0
+#define ngx_unlock_fd(fd) 0
+#define ngx_lock_fd_n "ngx_lock_fd()"
+#define ngx_unlock_fd_n "ngx_unlock_fd()"
+#endif
+
 
 static u_char  NGX_HTTP_VHOST_TRAFFIC_STATUS_DUMP_DATA_PAD[] = { 0x53, 0x54, 0x56 };
